iso4217_gen.go

```diff
@@ -123,16 +126,26 @@ func (cc CurrencyCode) Valid() bool {
 	fmt.Fprintln(&lookupBuffer, "var lookupTable = map[string]CurrencyCode{")
 
 	for i := range currencies {
-		code, name := currencies[i].Code, currencies[i].Name
-		if code == "" || name == "" {
-			fmt.Printf("SKIPPING: code=%q currency=%q\n", code, name)
+		code, name, microunit := currencies[i].Code, currencies[i].Name, currencies[i].MicroUnit
+		if code == "" || name == "" || microunit == "" {
+			fmt.Printf("SKIPPING: code=%q currency=%q microunit=%q\n", code, name, microunit)
 			continue
 		}
 		name = charCleaner.Replace(name)
+		microunit = charCleaner.Replace(microunit)
 		if _, exists := cs[code]; !exists {
 			cs[code] = true // mark as seen
 
-			fmt.Fprintf(&varBuffer, fmt.Sprintf(`  %s = CurrencyCode{Code: "%s", Name: "%s"}`+"\n", code, code, name))
+			decimalPlaces := -1
+			if microunit != "-" {
```

Comment:
"X currencies" such as gold, silver, etc have zero decimal places. They're units of troy ounces. We should return `DecimalPlaces: 0` in those cases. 

https://en.wikipedia.org/wiki/ISO_4217#X_currencies_(funds,_precious_metals,_supranationals,_other)